<a href="https://colab.research.google.com/github/Danae-Munoz/Analisis_Data_Mineria-de-datos/blob/main/Analisis_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#https://www.kaggle.com/datasets/robiulhasanjisan/student-exam-scores-and-study-habits-dataset?resource=download

In [ ]:
#importar librerias
import pandas as pd
import numpy as np

In [ ]:
#cargar dataset
df = pd.read_csv("/content/student_dataset.csv")

In [ ]:
#selecciona las columnas numericas y muestra info estadistica de ellas
df.describe().round()

,Horas_Estudio,Asistencia,Horas_Sueno,Puntuaciones_Anteriores,Sesiones_Tutoria,Puntuacion_Examen_Final
count,6607.0,6607.0,6607.0,6607.0,6607.0,6607.0
mean,20.0,80.0,7.0,75.0,1.0,60.0
std,6.0,12.0,1.0,14.0,1.0,9.0
min,1.0,60.0,4.0,50.0,0.0,30.0
25%,16.0,70.0,6.0,62.0,1.0,54.0
50%,20.0,80.0,7.0,75.0,1.0,60.0
75%,24.0,90.0,8.0,87.0,2.0,67.0
max,44.0,100.0,10.0,100.0,8.0,95.0


In [ ]:
#Mostrar 5 primeras filas del df
df.head()

,Horas_Estudio,Asistencia,Participacion_Parental,Acceso_a_Recursos,Actividades_ExtraCurriculares,Horas_Sueno,Puntuaciones_Anteriores,Nivel_Motivacion,Acceso_Internet,Sesiones_Tutoria,Puntuacion_Examen_Final
0,23.5,84.2,Low,Low,Yes,7.4,50.3,Medium,Yes,1,54.3
1,19.7,63.6,Medium,High,Yes,6.7,83.1,Low,Yes,2,59.4
2,24.4,98.4,Medium,Medium,Yes,8.0,63.5,Medium,Yes,0,63.8
3,29.6,88.6,Medium,High,Yes,6.3,57.5,Medium,Yes,2,61.0
4,19.1,91.5,High,Medium,Yes,6.2,66.0,Medium,Yes,2,60.7


In [ ]:
#sumar la cantidad de datos nulos por columna
df.isnull().sum()

,0
Hours_Studied,0
Attendance,0
Parental_Involvement,0
Access_to_Resources,0
Extracurricular_Activities,0
Sleep_Hours,0
Previous_Scores,0
Motivation_Level,0
Internet_Access,0
Tutoring_Sessions,0


In [ ]:
df = df.dropna()

In [ ]:
len(df.columns)

11

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6607 entries, 0 to 6606
Data columns (total 11 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Horas_Estudio                  6607 non-null   float64
 1   Asistencia                     6607 non-null   float64
 2   Participacion_Parental         6607 non-null   object 
 3   Acceso_a_Recursos              6607 non-null   object 
 4   Actividades_ExtraCurriculares  6607 non-null   object 
 5   Horas_Sueno                    6607 non-null   float64
 6   Puntuaciones_Anteriores        6607 non-null   float64
 7   Nivel_Motivacion               6607 non-null   object 
 8   Acceso_Internet                6607 non-null   object 
 9   Sesiones_Tutoria               6607 non-null   int64  
 10  Puntuacion_Examen_Final        6607 non-null   float64
dtypes: float64(5), int64(1), object(5)
memory usage: 567.9+ KB


**Esta data tiene 11 columnas y 6.607 datos**

In [ ]:
#tipos de datos de cada columna
df.dtypes

,0
Hours_Studied,float64
Attendance,float64
Parental_Involvement,object
Access_to_Resources,object
Extracurricular_Activities,object
Sleep_Hours,float64
Previous_Scores,float64
Motivation_Level,object
Internet_Access,object
Tutoring_Sessions,int64


In [ ]:
#trae toda las columnas del data frame
df.columns


Index(['Hours_Studied', 'Attendance', 'Parental_Involvement',
       'Access_to_Resources', 'Extracurricular_Activities', 'Sleep_Hours',
       'Previous_Scores', 'Motivation_Level', 'Internet_Access',
       'Tutoring_Sessions', 'Final_Exam_Score'],
      dtype='object')

**Variables categoricas:**
Parental_Involvement - Access_to_Resources - Extracurricular_Activities - Motivation_Level - Internet_Access - Tutoring_Sessions

In [ ]:
#Cambiamos nombres al español
columnas_espanol = {
    'Hours_Studied': 'Horas_Estudio',
    'Attendance': 'Asistencia',
    'Parental_Involvement': 'Participacion_Parental',
    'Access_to_Resources': 'Acceso_a_Recursos',
    'Extracurricular_Activities': 'Actividades_ExtraCurriculares',
    'Sleep_Hours': 'Horas_Sueno',
    'Previous_Scores': 'Puntuaciones_Anteriores',
    'Motivation_Level': 'Nivel_Motivacion',
    'Internet_Access': 'Acceso_Internet',
    'Tutoring_Sessions': 'Sesiones_Tutoria',
    'Final_Exam_Score': 'Puntuacion_Examen_Final'
}

df = df.rename(columns=columnas_espanol)


In [ ]:
df.columns

Index(['Horas_Estudio', 'Asistencia', 'Participacion_Parental',
       'Acceso_a_Recursos', 'Actividades_ExtraCurriculares', 'Horas_Sueno',
       'Puntuaciones_Anteriores', 'Nivel_Motivacion', 'Acceso_Internet',
       'Sesiones_Tutoria', 'Puntuacion_Examen_Final'],
      dtype='object')

In [ ]:
#mostrar valores unicos de la columna con cantidades
df['Participacion_Parental'].value_counts()

,count
Participacion_Parental,
Medium,3365
High,1968
Low,1274


**Participacion_Parental:**

Medium (Medio): Hay 3,365 registros. Es la categoría más común.


---


High (Alto): Hay 1,968 registros.


---


Low (Bajo): Hay 1,274 registros. Es la categoría menos frecuente.

In [78]:
#mostrar valores unicos de la columna con cantidades
df['Acceso_a_Recursos'].value_counts()

,count
Acceso_a_Recursos,
Medium,3349
High,1979
Low,1279


**Acceso_a_Recursos:**

Medium (Medio): 3,349 registros. La gran mayoría de las personas en tu base de datos tienen un acceso equilibrado o promedio a los recursos.


---


High (Alto): 1,979 registros. Es el segundo grupo más grande.


---


Low (Bajo): 1,279 registros. Es la minoría de tu muestra.

In [79]:
#mostrar valores unicos de la columna con cantidades
df['Actividades_ExtraCurriculares'].value_counts()

,count
Actividades_ExtraCurriculares,
Yes,3988
No,2619


**Actividades_ExtraCurriculares**

Yes (Sí): 3,988 personas sí participan en actividades fuera del horario de clases (deportes, música, talleres, etc.).


---


No: 2,619 personas no participan en ninguna actividad adicional.

In [ ]:
#mostrar valores unicos de la columna con cantidades
df['Nivel_Motivacion'].value_counts()

,count
Nivel_Motivacion,
Medium,3386
Low,1885
High,1336


In [ ]:
#mostrar valores unicos de la columna con cantidades
df['Acceso_Internet'].value_counts()

,count
Acceso_Internet,
Yes,6093
No,514


In [ ]:
#-----

In [ ]:
df['Actividades_ExtraCurriculares'].describe()

,Actividades_ExtraCurriculares
count,6607
unique,2
top,Yes
freq,3988
